test/monitoring/component-alerts_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build aws packet
```

Comment:
We don't run MetalLB on AWS.

Reply:
This file is generic to run tests for all the components not just metallb. So added a field in the test case struct to include the platforms list that test can specify.

Reply:
Oh, I don't like it at all, it seems super messy. Maybe we can use `BuildTags` from https://golang.org/pkg/go/build/#Context at least?

Reply:
Then we will have to write the test code `testComponentAlerts` for each platform. How do we make sure we don't repeat that?

Reply:
No, you can use  `t.Skip("reason")` : https://golang.org/pkg/testing/#hdr-Skipping if the build tag is not what you expect.
```go
if !buildTag("packet") {
  t.Skip("MetalLB should only be tested on AWS")
}
```

Reply:
But then again each test case will have to have a information on the supported platforms. Because this one test logic is same for all the components.

Reply:
So it turns out we cannot extract the build tags at runtime. Since it is a build time information, so using build tags to figure out what platform we are running on is not an option.

This needs to be done in some other easier way if using env var is repetition of the config.

Reply:
I think we are using build-time information for something that should be run-time information. IMO tags should represent things like the kind of tests (integration, unit, e2e, and so on) and for run-time information (like the platform we're running on) we should use env variables that we can check at run-time.

The platform is already passed as an env variable to Make, we're just placing that into build tags, which I don't think makes sense if we want to have shared logic in tests. I'd remove all build tags and use env variables, but that's a bigger task.

About the structuring of tests in this PR it seems OK to me.